sbi/inference/posteriors/likelihood_based_posterior.py

```diff
@@ -279,9 +284,12 @@ def np_potential(self, theta: np.array) -> ScalarFloat:
             Posterior log probability of the theta, $-\infty$ if impossible under prior.
         """
         theta = torch.as_tensor(theta, dtype=torch.float32)
-        log_likelihood = self.likelihood_nn.log_prob(
-            inputs=self.x.reshape(1, -1), context=theta.reshape(1, -1)
-        )
+        theta = ensure_theta_batched(theta)
+        num_batch = theta.shape[0]
+        x = ensure_x_batched(self.x).repeat(num_batch, 1)
+
+        with torch.set_grad_enabled(False):
```

Comment:
Why is this needed? Should be taken care of here: https://github.com/mackelab/sbi/blob/fc778f97b79cd13bc198bf14663e1725571c0ec9/sbi/inference/posteriors/base_posterior.py#L315

Reply:
It's not strictly necessary but won't hurt performance and prevents mistakes (e.g., unnecessarily tracking gradients when writing a new init strategy). I'd keep it in to make explicit that tracking gradients with np_potential is not supported

Reply:
Ok!